### PR TITLE
INGK-882 Autoscan with ixxat transciever connected raises an exception

### PR DESCRIPTION
**Pull request by MartinALL91**
### Description

Autoscan with IXXAT transciever connected raises an exception

Fixes # INGK-882

### Type of change

- Use the IXXAT CAN-FD implementation when scanning for drives.

### Tests

- Using an IXXAT transceiver scan for drives.
- Check that no unhandled exception occurs.

### Documentation

Please update the documentation.

- [x] Update docstrings of every function, method or class that change.
- [x] Add the changes at the `[Unreleased]` section of the [CHANGELOG](CHANGELOG.md).

### Code formatting

- [x] Use black package to format the code: `black -l 100 ingenialink tests`. It is recommended to configure the code editor to automatically format the code using black with a max length line of 100.

### Others

- [x] Set fix version field in the Jira issue.

### CODE REVIEW

File: ingenialink/canopen/network.py [hunk 1]
```diff
@@ -224,8 +224,9 @@ def scan_slaves(self) -> List[int]:
         is_connection_created = False
         if self._connection is None:
             is_connection_created = True
+            scan_with_ixxat = self.__device == CAN_DEVICE.IXXAT.value
             try:
-                self._setup_connection()
+                self._setup_connection(scan_with_ixxat)
```
Comment (pzinemanas) [inline ingenialink/canopen/network.py:229]: I think you should edit the scan_slaves_info method as well.

File: ingenialink/canopen/network.py [hunk 2]
```diff
@@ -422,17 +427,21 @@ def _reset_connection(self) -> None:
...
+        if self.__device == CAN_DEVICE.IXXAT.value:
+            connection_args["fd"] = True
         try:
-            self._connection.connect(
-                bustype=self.__device, channel=self.__channel, bitrate=self.__baudrate
-            )
+            self._connection.connect(**connection_args)
             for servo in self.servos:
                 servo.node = self._connection.add_node(servo.target)
                 servo.node.nmt.start_node_guarding(1)
```
Comment (GullinBustin) [inline ingenialink/canopen/network.py:442]: The node guarding period must be a constant, I know it is from another issue, but it must be fixed

**Review by GullinBustin — CHANGES_REQUESTED** [review]

File: ingenialink/canopen/network.py (hunk 2 above)
Comment (MartinALL91) [reply ingenialink/canopen/network.py]: Done

**Review by pzinemanas — APPROVED** [review]

File: ingenialink/canopen/network.py [hunk 3]
```diff
@@ -398,7 +421,10 @@ def _teardown_connection(self) -> None:
         if self._connection is None:
             logger.warning("Can not disconnect. The connection is not established yet.")
             return
-        self._connection.disconnect()
+        try:
+            self._connection.disconnect()
+        except VCIError as e:
+            logger.error(f"An exception occurred during the teardown connection. Exception: {e}")
```
Comment (GullinBustin) [inline ingenialink/canopen/network.py:427]: Why did you add this?

**Review by GullinBustin — APPROVED** [review]
